pkg/ddc/alluxio/shutdown.go

```diff
@@ -274,6 +279,14 @@ func (e *AlluxioEngine) destroyWorkers(expectedWorkers int32) (currentWorkers in
 			labelNames = append(labelNames, labelExclusiveName)
 		}
 
+		mutex.Lock()
```

Comment:
It Should use defer mutex.Unlock().

Reply:
@TrafalgarZZZ , Please take a look at the lock. Maybe the current lock scope is smaller than it should be.

Reply:
@cheyang, I Agree with you. The lock seems no effect to protect race condition. 
I suggest reusing the same lock defined in the scheduler(`AssignDatasetToNode` func).